metals/src/main/scala/scala/meta/internal/metals/codeactions/ImportMissingSymbol.scala

```diff
@@ -138,21 +137,50 @@ class ImportMissingSymbol(compilers: Compilers, buildTargets: BuildTargets)
           }
       )
       .map { actions =>
-        val deduplicated = actions.flatten
-          .groupBy(_.getTitle())
-          .map { case (_, actions) =>
-            val mainAction = actions.head
-            val allDiagnostics =
-              actions.flatMap(_.getDiagnostics().asScala).asJava
-            val edits = joinActionEdits(actions.toSeq)
-            mainAction.setDiagnostics(allDiagnostics)
-            mainAction
-              .setEdit(new l.WorkspaceEdit(Map(uri -> edits.asJava).asJava))
-            mainAction
-          }
-          .toSeq
-          .sorted
-        importMissingSymbols(deduplicated.toSeq)
+        val groupedByImported = actions
+          .filter(_.nonEmpty)
+          .groupBy(
+            _.head.getDiagnostics().asScala.headOption.collect {
+              case ScalacDiagnostic.SymbolNotFound(name) => name
+            }
+          )
+        val deduplicated = groupedByImported.flatMap {
+          case (None, actions) =>
+            actions
+          case (_, actions) =>
+            if (actions.length > 1) {
+              actions.minByOption(_.length) match {
```

Comment:
Shouldn't we return all that have min length? Not only first?

Reply:
Ok, so I reworked that, I think it looks nicer. I now do a intersect of all possibilities, basically what I should have done at the beginning.